pkgs/development/libraries/protobuf/generic-v3.nix

```diff
@@ -1,13 +1,12 @@
 { stdenv
 , fetchFromGitHub
 , autoreconfHook, zlib, gmock, which, buildPackages
-, useExternalProtoc
-, buildProtobuf ? null
 , version, sha256
 , ...
 }:
 
-stdenv.mkDerivation rec {
+let
+mkProtobufDerivation = buildProtobuf: stdenv: stdenv.mkDerivation rec {
```

Comment:
I don't think you need a function here? Can't you just do `let buildProtobuf = if stdenv.hostPlatform != stdenv.buildPlatform then buildPackages.protobuf else null; in stdenv.mkDerivation rec {…}`? I'm not sure about the exact recursion implications of that, but I think it should work.

Reply:
Good call!